### ISSUE TITLE
Data Summary Feature

### ISSUE THREAD

**Issue by ccoffrin** [lanl-ansi/InfrastructureModels.jl#2]
PowerModel's `summary` feature is a fairly generic function which might be applicable to a wide class of data,
https://github.com/lanl-ansi/PowerModels.jl/blob/master/src/core/data.jl#L135

We should check if it can be easily generalized to work with GasModels and/or WaterModels, and if-so move it into this package.

@rb004f

**Comment by rb004f** [comment]
Agreed, this looks like something that would be useful to a wide range for models.
